pydantic_ai_slim/pydantic_ai/output.py

```diff
@@ -304,6 +305,19 @@ def StructuredDict(
     """
     json_schema = _utils.check_object_json_schema(json_schema)
 
+    # If the schema contains $defs, inline them to avoid issues with pydantic's
+    # JSON schema generator (Issue #2466)
+    if '$defs' in json_schema:
+        from .profiles import InlineDefsJsonSchemaTransformer
+
+        try:
+            transformer = InlineDefsJsonSchemaTransformer(json_schema)
+            json_schema = transformer.walk()
+        except UserError:
+            # If the transformer can't resolve refs (e.g., missing definitions),
+            # keep the original schema unchanged
```

Comment:
Let's raise a descriptive error instead of continuing and hitting the confusing Pydantic error